internal/store/plugin.go

```diff
@@ -62,3 +69,48 @@ func (store *Store) GetPlugins(pluginFilter *model.PluginFilter) ([]*model.Plugi
 
 	return plugins[start:end], nil
 }
+
+func (store *Store) getPlugins(serverVersion string) ([]*model.Plugin, error) {
+	var result []*model.Plugin
+
+	for _, plugin := range store.plugins {
+		p, err := getPlugin(serverVersion, plugin)
+		if err != nil && err != ErrNotFound {
+			return nil, errors.Wrapf(err, "failed to get plugin")
+		} else if err != ErrNotFound {
+			result = append(result, p)
+		}
+	}
+
+	return result, nil
+}
+
+// getPlugin gets the first plugin from the sorted pluginVersions slice that satisfies serverVersion.
+func getPlugin(serverVersion string, pluginVersions pluginVersions) (*model.Plugin, error) {
```

Comment:
Would it make sense to define `getPlugin` on `store *Store`? If not, would it make sense to demote `getPlugins` to a plain method and pass in `plugins` instead?